4-Create-Data-Viz-Dashboard.md

```diff
@@ -5,44 +5,41 @@
 1. [Next step](#next-step)
 
 ## You have data, now what?
-In [Step 3](3-Make-API-Calls.md) you fetched discussion data from the `/getDiscussions` endpoint created and set the `discussion` variable. Now we're passing this data from the `App` component to some child [React component](https://reactjs.org/docs/components-and-props.html) that will render the heatmap.
 
-There are two child components that we've prebuilt for you: `Discussion` and `Heatmap`.
+### Discussion
 
-The discussion data from `App` will be sent to `Discussion`, which will modify the data and send it to `Heatmap`, which will finally render the heatmap.
+In [Step 3](3-Make-API-Calls.md) you fetched discussion data from the `/getDiscussions` endpoint and set the `discussion` variable from within `Discussion`.
 
-## Discussion component
-In `App`, you'll see the following line near the bottom:
+We've written the following `useEffect` from within `Discussion`:
 
 ```js
-<Discussion data={discussion} />
+useEffect(() => {
+  if (discussion.length > 0) {
+    const discussionTimestamps = discussion
+      .map(discussion => discussion.timestamp)
+
+    setTimestamps(discussionTimestamps)
+  }
+}, [discussion])
 ```
+It checks to see if discussion data has been set - if so, it extracts the timestamp property and sets the `timestamps` variable. Why do this instead of just sending the data as is to `Heatmap`? Well, we designed `Heatmap` to only rely on `timestamps`, and it's better practice to send child components just the data it requires instead of the entire kitchen sink.
```

Comment:
I like that you've explained and encouraged best practices! Also forces participants to stop and really think about what's going on 👍